fendermint/ipc/src/pof.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+//! IPC proof of finality related functions
+
+use crate::Config;
+use crate::{BlockHeight, ParentViewProvider};
+use anyhow::anyhow;
+use ipc_sdk::cross::CrossMsg;
+use ipc_sdk::ValidatorSet;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+/// The proof for POF.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct IPCParentFinality {
+    /// The latest chain height
+    pub height: u64,
+    /// The block hash. For FVM, it is a Cid. For Evm, it is bytes32.
+    pub block_hash: Vec<u8>,
+    /// new top-down messages finalized in this PoF
+    pub top_down_msgs: Vec<CrossMsg>,
+    /// latest configuration information from the parent.
+    pub config: ValidatorSet,
+}
+
+/// The parent finality information provider. It provides the latest finality information about the
+/// parent. Also it perform validation on incoming finality.
+#[derive(Clone)]
+pub struct ParentFinalityProvider<T> {
+    config: Config,
```

Comment:
Case in point: here's another `config` field, less than 10 lines away, with a completely different meaning.